org.eclipse.xtext/src/org/eclipse/xtext/formatting2/regionaccess/internal/HiddenRegionPartAssociator.java

```diff
@@ -25,12 +25,15 @@
 	public void associate(IHiddenRegion region, BiConsumer<IHiddenRegionPart, HiddenRegionPartAssociation> handler) {
 		List<IHiddenRegionPart> parts = region.getParts();
 		if (!region.isMultiline()) {
+			// when there is no line-wrap between two semantic tokens, assume that comments belong to the CONTAINER
+			// rather than one of the surrounding semantic tokens.
 			for (IHiddenRegionPart p : parts) {
 				handler.accept(p, HiddenRegionPartAssociation.CONTAINER);
 			}
 		} else {
 			int i = 0;
 			if (region.getPreviousSemanticRegion() != null) {
+				// collect whitespace and comments from the same line as belonging to the PREVIOUS semantic token
```

Comment:
typo? are?

Reply:
no, both whitespace AND comments are collected.

Reply:
no i means `as` sounds wired in the sentence

Reply:
I see. Same as in `they're treated as...`